internal/slicer/slicer_test.go

```diff
@@ -1737,6 +1737,94 @@ var slicerTests = []slicerTest{{
 		"/dir/file2": "file 0644 28121945 <2> {test-package2_myslice}",
 		"/hardlink2": "file 0644 28121945 <2> {test-package2_myslice}",
 	},
+}, {
+	summary: "Mutations for hard links are forbidden",
+	slices: []setup.SliceKey{
+		{"test-package", "myslice"}},
+	pkgs: []*testutil.TestPackage{{
+		Name: "test-package",
+		Data: testutil.MustMakeDeb([]testutil.TarEntry{
+			testutil.Dir(0755, "./"),
+			testutil.Dir(0755, "./dir/"),
+			testutil.Reg(0644, "./dir/file", "text for file"),
+			testutil.Hln(0644, "./hardlink", "./dir/file"),
+		}),
+	}},
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					contents:
+						/dir/file:
+						/hardlink: {mutable: true}
+					mutate: |
+						content.write("/hardlink", "new content")
+		`,
+	},
+	error: `slice test-package_myslice: cannot mutate a hard link: /hardlink`,
+}, {
+	summary: "Hard links can be marked as mutable, but not mutated",
+	slices: []setup.SliceKey{
+		{"test-package", "myslice"}},
+	pkgs: []*testutil.TestPackage{{
+		Name: "test-package",
+		Data: testutil.MustMakeDeb([]testutil.TarEntry{
+			testutil.Dir(0755, "./"),
+			testutil.Dir(0755, "./dir/"),
+			testutil.Reg(0644, "./dir/file", "text for file"),
+			testutil.Hln(0644, "./hardlink", "./dir/file"),
+		}),
+	}},
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					contents:
+						/dir/file:
+						/hardlink: {mutable: true}
+		`,
+	},
+	filesystem: map[string]string{
+		"/dir/":     "dir 0755",
+		"/dir/file": "file 0644 28121945",
+		"/hardlink": "file 0644 28121945",
+	},
+	manifestPaths: map[string]string{
+		"/dir/file": "file 0644 28121945 <1> {test-package_myslice}",
+		"/hardlink": "file 0644 28121945 <1> {test-package_myslice}",
+	},
+}, {
+	summary: "UntilMutate for hard links are allowed",
```

Comment:
I don't think we need to check this, there is nothing special about hard links and until mutate right?